designer/src/components/condition.tsx

```diff
@@ -423,11 +628,174 @@ const FieldConditionControl = (props: ConditionProps) => {
     }
   };
 
-  const addCondition = () => {
+  const getFieldLabel = (f: FieldType) =>
+    f?.['component-parameters']?.InputLabelProps?.label ||
+    f?.['component-parameters']?.name ||
+    '<unlabeled>';
+
+  /* Checks if a field has predefined options */
+  const isPredefinedOptions = (fieldDef: FieldType | null): boolean => {
+    if (!fieldDef) return false;
+    return ['Select', 'RadioGroup', 'MultiSelect', 'Checkbox'].includes(
+      fieldDef['component-name']
+    );
+  };
+
+  /* Filter the allowed operators based on field type */
+  const getAllowedOperators = (fieldDef: FieldType | null) => {
+    return isPredefinedOptions(fieldDef)
+      ? ['equal', 'not-equal']
+      : ['equal', 'not-equal', 'greater', 'less', 'contains', 'regex'];
+  };
+
+  const renderValueEditor = (fieldDef: FieldType) => {
+    const cName = fieldDef['component-name'];
+    const params = fieldDef['component-parameters'] || {};
+    const possibleOptions = params.ElementProps?.options || [];
+
+    const isValidOption = possibleOptions.some(
+      (opt: any) => opt.value === condition.value
+    );
+
+    switch (cName) {
+      case 'Select':
+      case 'RadioGroup': {
+        return (
+          <FormControl sx={{minWidth: 200}} error={!isValidOption}>
+            <InputLabel>Value</InputLabel>
+            <Select
+              label="Value"
+              value={isValidOption ? condition.value : (condition.value ?? '')} // Show invalid value
+              onChange={e => updateValue(e.target.value)}
+            >
+              {possibleOptions.map((opt: any) => (
+                <MenuItem key={opt.value} value={opt.value}>
+                  {opt.label}
+                </MenuItem>
+              ))}
+            </Select>
+            {!isValidOption && (
+              <div style={{color: 'red', fontSize: '12px'}}>
+                Invalid value: "{String(condition.value)}"
+              </div>
+            )}
+          </FormControl>
+        );
+      }
+
+      case 'MultiSelect': {
+        const selectedValues = Array.isArray(condition.value)
+          ? condition.value
+          : [];
+
+        return (
+          <FormControl
+            sx={{minWidth: 200}}
+            error={
+              !selectedValues.every(v =>
+                possibleOptions.some((opt: any) => opt.value === v)
+              )
+            }
+          >
+            <InputLabel>Value</InputLabel>
+            <Select
+              multiple
```

Comment:
You're allowing me to choose multiple values here but then the condition won't work since it will say 'x equals [y, z]' or 'x contains [y, z]', I need an explicit OR if I want to check for alternate values - unless we have a 'contains one of' operator, which we could have...